modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/components/A11y/A11y.tsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import './A11y.scss';
+
+import ClayIcon from '@clayui/icon';
+import ClayList from '@clayui/list';
+import ClayPopover from '@clayui/popover';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import useA11y from '../../hooks/useA11y';
+
+import type {Violation as TViolation} from '../../hooks/useA11y';
+import type {A11yCheckerOptions} from './A11yChecker';
+
+const rectAttrs: Array<keyof DOMRect> = [
+	'bottom',
+	'height',
+	'left',
+	'right',
+	'top',
+	'width',
+];
+
+interface IRectState {
+	rect: DOMRect | undefined;
+	hasRectChanged: boolean;
+}
+
+const rectChanged = (a: DOMRect = {} as DOMRect, b: DOMRect = {} as DOMRect) =>
```

Comment:
nit: Reading these types seems a bit odd. You are saying `a` is a `DOMRect` type, but then you are immediately assigning it to an empty object and casting it. It might be slightly more clear to create a "dummy" DOMRect value.

```js
const DOMRectStub = {} as DOMRect;

const rectChanged = (a = DOMRectStub, b = DOMRectStub) => {

```